src/unix/cygwin/mod.rs

```diff
@@ -0,0 +1,3137 @@
+
+pub const SCHED_OTHER: c_int = 3;
+pub const SCHED_FIFO: c_int = 1;
+pub const SCHED_RR: c_int = 2;
+
+pub const PTHREAD_COND_INITIALIZER: pthread_cond_t = 21 as *mut _;
+pub const PTHREAD_CREATE_DETACHED: c_int = 1;
+pub const PTHREAD_CREATE_JOINABLE: c_int = 0;
+pub const PTHREAD_MUTEX_RECURSIVE: c_int = 0;
+pub const PTHREAD_MUTEX_ERRORCHECK: c_int = 1;
+pub const PTHREAD_MUTEX_NORMAL: c_int = 2;
+pub const PTHREAD_MUTEX_DEFAULT: c_int = PTHREAD_MUTEX_NORMAL;
+pub const PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP: pthread_mutex_t = 18 as *mut _;
+pub const PTHREAD_ERRORCHECK_MUTEX_INITIALIZER_NP: pthread_mutex_t = 20 as *mut _;
+pub const PTHREAD_MUTEX_INITIALIZER: pthread_mutex_t = 19 as *mut _;
+pub const PTHREAD_PROCESS_SHARED: c_int = 1;
+pub const PTHREAD_PROCESS_PRIVATE: c_int = 0;
+pub const PTHREAD_RWLOCK_INITIALIZER: pthread_rwlock_t = 22 as *mut _;
+
+pub const LITTLE_ENDIAN: c_int = 1234;
+pub const BIG_ENDIAN: c_int = 4321;
+
+pub const TCP_NODELAY: c_int = 1;
+pub const TCP_KEEPIDLE: c_int = 3;
+pub const TCP_MAXSEG: c_int = 4;
+pub const TCP_QUICKACK: c_int = 12;
+pub const TCP_USER_TIMEOUT: c_int = 14;
+pub const TCP_FASTOPEN: c_int = 15;
+pub const TCP_KEEPCNT: c_int = 16;
+pub const TCP_KEEPINTVL: c_int = 17;
+
+pub const WINDOWS_POST: Ioctl = 0;
+pub const WINDOWS_SEND: Ioctl = 1;
+pub const WINDOWS_HWND: Ioctl = 2;
+
+pub const MOUNT_TEXT: c_uint = 0x01;
+pub const MOUNT_SYSTEM: c_uint = 0x08;
+pub const MOUNT_EXEC: c_uint = 0x10;
+pub const MOUNT_CYGDRIVE: c_uint = 0x20;
+pub const MOUNT_CYGWIN_EXEC: c_uint = 0x40;
+pub const MOUNT_SPARSE: c_uint = 0x80;
+pub const MOUNT_NOTEXEC: c_uint = 0x100;
+pub const MOUNT_DEVFS: c_uint = 0x200;
+pub const MOUNT_PROC: c_uint = 0x400;
+pub const MOUNT_RO: c_uint = 0x1000;
+pub const MOUNT_NOACL: c_uint = 0x2000;
+pub const MOUNT_NOPOSIX: c_uint = 0x4000;
+pub const MOUNT_OVERRIDE: c_uint = 0x8000;
+pub const MOUNT_IMMUTABLE: c_uint = 0x10000;
+pub const MOUNT_AUTOMATIC: c_uint = 0x20000;
+pub const MOUNT_DOS: c_uint = 0x40000;
+pub const MOUNT_IHASH: c_uint = 0x80000;
+pub const MOUNT_BIND: c_uint = 0x100000;
+pub const MOUNT_USER_TEMP: c_uint = 0x200000;
+pub const MOUNT_DONT_USE: c_uint = 0x80000000;
+
+pub const _POSIX_VDISABLE: cc_t = 0;
+
+pub const GRND_NONBLOCK: c_uint = 0x1;
+pub const GRND_RANDOM: c_uint = 0x2;
+
+pub const _IONBF: c_int = 2;
+pub const BUFSIZ: c_int = 1024;
+
+pub const POSIX_SPAWN_RESETIDS: c_int = 0x01;
+pub const POSIX_SPAWN_SETPGROUP: c_int = 0x02;
+pub const POSIX_SPAWN_SETSCHEDPARAM: c_int = 0x04;
+pub const POSIX_SPAWN_SETSCHEDULER: c_int = 0x08;
+pub const POSIX_SPAWN_SETSIGDEF: c_int = 0x10;
+pub const POSIX_SPAWN_SETSIGMASK: c_int = 0x20;
+
+// intentionally not public, only used for fd_set
+cfg_if! {
+    if #[cfg(target_pointer_width = "32")] {
+        const ULONG_SIZE: usize = 32;
+    } else if #[cfg(target_pointer_width = "64")] {
+        const ULONG_SIZE: usize = 64;
+    } else {
+        // Unknown target_pointer_width
+    }
+}
```

Comment:
Could these use `core::mem::size_of::<c_ulong>()` instead?

Reply:
The other targets use this kind of definition.

Reply:
That probably predates const `size_of`. We don't need to support versions that old anymore so this can use the new version, the others will get cleaned up at some point.